runtime/j9vm/javanextvmi.cpp

```diff
@@ -326,34 +327,41 @@ JVM_VirtualThreadMountEnd(JNIEnv *env, jobject thread, jboolean firstMount)
 				J9VMJDKINTERNALVMCONTINUATION_VMREF(currentThread, continuationObj));
 	}
 
+	/* On some occassions, J9AllocateObject acquires exclusive VM access. It is invoked without acquiring
+	 * liveVirtualThreadListMutex to prevent a deadlock.
+	 */
+	if (NULL == vm->liveVirtualThreadList) {
+		Assert_SC_true(firstMount);
+		J9Class *virtualThreadClass = J9OBJECT_CLAZZ(currentThread, J9_JNI_UNWRAP_REFERENCE(thread));
+		J9MemoryManagerFunctions *mmFuncs = vm->memoryManagerFunctions;
+
+		/* Allocate empty virtual thread and create a global reference to it as root for the linked list.
+		 * This prevents the root reference from becoming stale if the GC moves the object.
+		 */
+		rootVirtualThread = mmFuncs->J9AllocateObject(currentThread, virtualThreadClass, J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
+		if (NULL == rootVirtualThread) {
+			vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
```

Comment:
This is a heap OOM, not native.

Reply:
Updated both locations.

Reply:
Will the other `j9jni_createGlobalRef` location also be a heap OOM? If not, I will revert that change.

Reply:
Yes - refs are allocated in native memory.